uwblueprint/sistema
# New User Management

# Notion ticket link



[New User Storage](https://www.notion.so/uwblueprintexecs/Sistema-03b6ce1c4bc447b5945ab8b79cb00ea5?p=b37f8e4e76444d2d8c0745da42a09b0f&pm=s)



## Implementation description
- Adjusted Auth.ts code to add the User when they sign in.
- Added a new Table for Accounts to hold the account information -- leaves room for the future if we want to also add Sessions & Tokens
- Added default values to the users -> auto-roled as a "Teacher",  default status is "Invited" & default numofAbsences are "10"
 



## Steps to test

1. Run Docker -- self-explanatory
2. On localhost:3000, sign in with Google & log in with any account (no restrictions currently set)
3. Create a new Terminal (aside from the Docker one), and run the following commands:
`docker exec -it sistema-db-1 psql -U sistema -d sistema`
`SELECT * FROM public."User";`
Ensure your account comes up on the bottom & all columns are filled.

From there, verify that Accounts are also filled. Run `SELECT * FROM public."Account";`
Ensure your account is filled & all columns are filled.




## What should reviewers focus on?

- Ensure the the users are added to BOTH "User" table and "Account" table
- Ensure that you can get to the homepage
- Ensure no errors persist on Docker 

## Checklist

- [ X] My PR name is descriptive and in imperative tense
- [ X] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [X ] I have run the appropriate linter(s)
- [ X] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
